lib/std/Thread/Futex.zig

```diff
@@ -66,7 +66,7 @@ pub fn wake(ptr: *const atomic.Value(u32), max_waiters: u32) void {
     Impl.wake(ptr, max_waiters);
 }
 
-const Impl = if (builtin.single_threaded)
+const Impl = if (builtin.single_threaded or builtin.os.tag == .uefi)
```

Comment:
Should UEFI be treated as a proper single-threaded target?

https://github.com/ziglang/zig/blob/82f35c51866826fb71eb6f27d9210bb738d602e9/src/target.zig#L66-L81

Reply:
Yes, I just didn't know where the source was which defined that heh.

Reply:
I've updated this and changed the `defaultSingleThreaded` for UEFI.

Reply:
Out of curiosity, is UEFI not _always_ single threaded instead of just default single threaded?

Reply:
You technically could implement a form of multithreading, just like with wasm. It's just `builtin.single_threaded` will default to `true` when `single_threaded` is not changed in your build. You probably could write a custom scheduler + mutltithreading stuff and use `root.os` stuff to get it working but no implementation will be made for upstream.

Reply:
I learned somewhat recently that UEFI does have `EFI_MP_SERVICES_PROTOCOL` so there is a way to do multiprocessing but not multithreading but likely is still out of the scope.